### DISCUSSION TITLE
How can I use min and optional method together?

### DISCUSSION THREAD

**Discussion by lgj9172** [colinhacks/zod#1251] [Q&A] [answered]
I should make restricted input but it can be also optional.
So I do like this.

```

z.union([
      z.string()
        .min(2, TOO_SHORT_COMPANY)
        .max(40, TOO_LONG_COMPANY)
        .regex(/^[0-9a-zA-Z]/,ALPHABET_NUMBER_ONLY_COMPANY),
      z.string().optional(),
    ])

```

But min method is passed.(not work)
Could any body help me to do this properly?

**Comment by Shumuu** [comment, marked as answer]
This works!

```
z.optional(z.string()
.min(2, TOO_SHORT_COMPANY)
.max(40, TOO_LONG_COMPANY)
.regex(/^[0-9a-zA-Z]/,ALPHABET_NUMBER_ONLY_COMPANY))
```

[CodeSandBox](https://codesandbox.io/s/loving-benz-gu8ofd?file=/index.ts)

**Reply by lgj9172** [reply]
I am so appreciate for your answer. But it is just work about min and after. Not optional.

**Reply by lgj9172** [reply]
I think it is because input element have ""(Empty string). Should I allow empty string instead of optional?
I get a idea from your codesandbox. It gives me help a lot.

**Reply by lgj9172** [reply]
OMG I got an answer what I want.

```
z.union([
  z
    .string()
    .min(2, TOO_SHORT_COMPANY)
    .max(40, TOO_LONG_COMPANY)
    .regex(/^[0-9a-zA-Z]/, ALPHABET_NUMBER_ONLY_COMPANY),
  z.literal(''),
])
```

You let me know that empty string and optional is different and how to use optional conditionally.
I am so thank you about your answer and code.
